lib/parameterized-sql.js

```diff
@@ -39,7 +39,9 @@ function ParameterizedSQL(sql, params) {
     'The number of ? (' + (parts.length - 1) +
     ') in the sql (' + this.sql + ') must match the number of params (' +
     this.params.length +
-    ') ' + this.params);
+    ') ' + this.params.map(p => {
+      return p !== Object(p)? p : `<${typeof p}>`;
```

Comment:
This will affect `Date` instances too. Can we use [`Buffer.isBuffer`](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_isbuffer_obj) check instead?

```suggestion
      return Buffer.isBuffer(p) ? '<Buffer>' : p;
```

It may be a good idea to use [`util.inspect`](https://nodejs.org/api/util.html#util_util_inspect_object_showhidden_depth_colors) to convert values to strings in a way that allows readers to distinguish between a number `123` and a string `'123`'. This function already handles large Buffer values:

```
<Buffer 7b 0a 20 20 22 6e 61 6d 65 22 3a 20 22 6c 6f 6f 70 62 61 63 6b 2d 6e 65 78 74 22 2c 0a 20 20 22 72 65 70 6f 73 69 74 6f 72 79 22 3a 20 7b 0a 20 20 20 ... 4171 more bytes>
```

Reply:
This still needs to be optimized further.  It is big performance hit. as params is always stringified, even if the assertion is true.